lib/pinterest/omniauth.rb

```diff
@@ -30,11 +31,56 @@ def ssl?
         true
       end
 
+      def build_access_token
+        token = super
+
+        if token.expires_at == 0
+          ::OAuth2::AccessToken.new(token.client, token.token,
+            token.params.reject{|k,_| [:refresh_token, :expires_in, :expires_at, :expires].include? k.to_sym}
+          )
+        else
+          token
+        end
+      end
+
       def raw_info
-        fields = 'first_name,id,last_name,url,account_type,username,bio,image'
         @raw_info ||=
-          access_token.get("/v1/me/?fields=#{fields}").parsed['data']
+          access_token.get("users/me/").parsed['data']
+      end
+    end
+  end
+end
+
+module OAuth2
+  # The OAuth2::Client class
+  class Client # rubocop:disable Metrics/ClassLength
+
+    # Initializes an AccessToken by making a request to the token endpoint
+    #
+    # @param [Hash] params a Hash of params for the token endpoint
+    # @param [Hash] access token options, to pass to the AccessToken object
+    # @param [Class] class of access token for easier subclassing OAuth2::AccessToken
+    # @return [AccessToken] the initialized AccessToken
+    def get_token(params, access_token_opts = {}, access_token_class = AccessToken) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
```

Comment:
Had to overwrite this since Pinterest returned their response data in a different format.